x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/queries/SemanticMatchQueryRewriteInterceptor.java

```diff
@@ -63,14 +58,15 @@ public QueryBuilder rewrite(QueryRewriteContext context, QueryBuilder queryBuild
                 BoolQueryBuilder boolQueryBuilder = new BoolQueryBuilder();
                 for (String inferenceIndexName : inferenceIndices) {
                     // Add a separate clause for each semantic query, because they may be using different inference endpoints
+                    // TODO - consolidate this to a single clause once the semantic query supports multiple inference endpoints
                     boolQueryBuilder.should(
                         createSemanticSubQuery(inferenceIndexName, matchQueryBuilder.fieldName(), (String) matchQueryBuilder.value())
                     );
                 }
```

Comment:
Match query support with multiple inference endpoints is already broken (due to the index filter clause running as a post-filter), so we can consolidate to a single clause now.